subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -992,6 +992,14 @@ private boolean isJava7Home(String path){
         return false;
     }
 
+    private boolean isJava7DeprecationTruncatedDaemonLog(List<String> lines, int i, String line) {
+        // for forkingIntegrationTests running on Java 7, daemon exit log will fail the tests with following stdout:
+        // ----- Last  20 lines from daemon log file - daemon-6753.out.log -----
+        //    at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
+        //    at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
+        //    ...
+        return line.startsWith("----- Last  ") && i < lines.size() - 1 && STACK_TRACE_ELEMENT.matcher(lines.get(i + 1)).matches();
+    }
```

Comment:
:x: This does not look right. The daemon should not exit with an exception. Can you point me at the test(s) that are/were failing?

Reply:
This has nothing to do with the daemon disappear issue. This is to fix this case:

On Java 7, in forkingIntegTest, deprecation warning will be printed into daemon log, then truncated and displayed on forkingIntegTest's stdout, which breaks tests with 'unexpected stacktrace'. This code is just a workaround to avoid modifying dozens of forkingIntegTests and should be removed in subsequent development.

Reply:
Alright. I got that wrong. Thanks for clarifying.